extensions/resource-deployment/src/ui/validation/validations.ts

```diff
@@ -94,17 +110,16 @@ export class RegexValidation extends Validation {
 		return this._regex;
 	}
 
-	constructor(validation: RegexValidationInfo, valueGetter: ValueGetter) {
-		super(validation, valueGetter);
+	constructor(validation: RegexValidationInfo, validationMessageUpdater: onValidation, valueGetter: ValueGetter) {
+		super(validation, validationMessageUpdater, valueGetter);
 		throwUnless(validation.regex !== undefined);
 		this._regex = (typeof validation.regex === 'string') ? new RegExp(validation.regex) : validation.regex;
 	}
 
 	async validate(): Promise<ValidationResult> {
-		const value = await this.getValue();
-		const isValid = value === undefined
-			? false
-			: this.regex.test(value.toString());
+		const value = (await this.getValue())?.toString();
+		const isValid = this.isUndefinedOrEmpty(value) ? true : this.regex.test(value!);
```

Comment:
For these shouldn't we only be allowing undefined/empty values if the field isn't required? If it is required then we should be running that through the regex validation.

Reply:
Ping

Reply:
If the field is required then while this validation succeeds the builtin required validaiton fails and when a value has been entered by the user then the value is no longer undefined or empty and hence this validation comes into full force. I think I should document this on top of the Validation class as it applies to all validations.